/*
Name: Ambient Ping Pong
ID: D3F95035-B1A9-C1DE-A6D3-54E421E8567C
original size : uint8_t preset_7[1544]
 */

// [START declaration]
syntax = "proto2";
package presets;
// [END declaration]

// [START messages]
// ======== data.json ========
message CParam {
    optional int32 id = 1;
    optional float value = 2;
}

message CSigPath {
    message CBlocks {
        message CItem {
            optional string id = 1;
            optional bool isActive = 2;
            optional string name = 3;
            repeated CParam params = 4;
            optional string selected = 5;
        }

        repeated CItem items = 1;
    }

    message CMetronome {
        optional int32 bpm = 1;
        optional bool isActive = 2;
    }

    message CNoisegate {
        optional string id = 1;
        optional bool isActive = 2;
        optional string name = 3;
        repeated CParam params = 4;
        optional string selected = 5;
    }

    optional CBlocks blocks = 1;
    optional CMetronome metronome = 2;
    optional CNoisegate noisegate = 3;
}

message CPresetDataJson {
    optional string panelStyle = 1;
    optional CSigPath sigPath = 2;
}
// ===========================

// [END messages]
